### PR TITLE
[FLINK-30693][sql-gateway] Introduce RowFormat to SQL Gateway

### PR DESCRIPTION
**Pull request by yuzelin**
## What is the purpose of the change

Currently, the SQL gateway only sends JSON-formatted result to REST endpoint. So this PR introduces RowFormat to control the result format.

## Brief change log

  - Add `RowFormat` to the SQL gateway.
  - Add codes in SQL gateway REST API to handle the `RowFormat`.

## Verifying this change

This change added tests and can be verified as follows:

  - `ResultInfoJsonSerDeTest`
  - `SqlGatewayRestEndpointStatementITCase`
  - `StatementRelatedITCase`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: yes
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented?  JavaDocs

### CODE REVIEW

**Review by fsk119 — COMMENTED** [review]
@yuzelin Could you take some time to check the update? Because Spring Festival is coming, I  just add commits to reduce the communication time.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/message/statement/FetchResultsRowFormatQueryParameter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.flink.runtime.rest.messages.MessageQueryParameter;
+import org.apache.flink.table.gateway.rest.util.RowFormat;
+
+/**
+ * A {@link MessageQueryParameter} that parses the 'rowFormat' query parameter for fetching results.
+ */
+public class FetchResultsRowFormatQueryParameter extends MessageQueryParameter<RowFormat> {
+
+    public static final String KEY = "rowFormat";
+
```
Comment (yuzelin) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/message/statement/FetchResultsRowFormatQueryParameter.java:30]: Seems other REST APIs use `_`, like `SessionHandleIdPathParameter.KEY`

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/handler/statement/FetchResultsHandler.java [hunk 2]
```diff
@@ -30,64 +32,82 @@
...
+                            FetchResultsResponseBody,
+                            FetchResultsMessageParameters>
                     messageHeaders) {
         super(service, responseHeaders, messageHeaders);
     }
 
     @Override
     protected CompletableFuture<FetchResultsResponseBody> handleRequest(
-            SqlGatewayRestAPIVersion version, @Nonnull HandlerRequest<EmptyRequestBody> request) {
+            SqlGatewayRestAPIVersion version, @Nonnull HandlerRequest<EmptyRequestBody> request)
```
Comment (yuzelin) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/handler/statement/FetchResultsHandler.java:68]: How about delete the `@Nonnull` (and also delete it from it's parent)?

**Review by yuzelin — COMMENTED** [review]
I find a few points that may be improved.

**Review by fsk119 — APPROVED** [review]
Thanks for your review and update. LGTM.

Merging...

**Pull request closed without merging**